pinot-common/src/main/java/org/apache/pinot/common/config/provider/TableCache.java

```diff
@@ -476,6 +581,49 @@ public synchronized void handleDataDeleted(String path) {
     }
   }
 
+  private class ZkLogicalTableChangeListener implements IZkChildListener, IZkDataListener {
+
+    @Override
+    public synchronized void handleChildChange(String path, List<String> logicalTableNames) {
+      if (CollectionUtils.isEmpty(logicalTableNames)) {
+        return;
+      }
+
+      // Only process new added logical tables. Changed/removed logical tables are handled by other callbacks.
+      List<String> pathsToAdd = new ArrayList<>();
+      for (String logicalTableName : logicalTableNames) {
+        if (!_logicalTableNameMap.containsKey(logicalTableName)) {
+          pathsToAdd.add(LOGICAL_TABLE_PATH_PREFIX + logicalTableName);
+        }
+      }
+      if (!pathsToAdd.isEmpty()) {
+        addLogicalTables(pathsToAdd);
+      }
+      notifyLogicalTableChangeListeners();
+    }
+
+    @Override
+    public synchronized void handleDataChange(String path, Object data) {
+      if (data != null) {
+        ZNRecord znRecord = (ZNRecord) data;
+        try {
+          putLogicalTable(znRecord);
+        } catch (Exception e) {
+          LOGGER.error("Caught exception while refreshing logical table for ZNRecord: {}", znRecord.getId(), e);
+        }
+        notifyLogicalTableChangeListeners();
+      }
+    }
+
+    @Override
+    public synchronized void handleDataDeleted(String path) {
+      // NOTE: The path here is the absolute ZK path instead of the relative path to the property store.
+      String logicalTableName = path.substring(path.lastIndexOf('/') + 1);
+      removeLogicalTable(LOGICAL_TABLE_PATH_PREFIX + logicalTableName);
+      notifyLogicalTableChangeListeners();
+    }
+  }
+
   private static class TableConfigInfo {
```

Comment:
We will likely need a entry similar to this for logical table. These are all query related properties.

(minor) Since logical table config handling is very similar to table config handling, consider putting all new methods next to table config methods for readability.

Reply:
Can this be taken up in a separate PR for https://github.com/apache/pinot/issues/15607 ?

Reply:
I have got the basic structure for this. So let go ahead with it. We can extend it further when we work on the query options.

Reply:
I checked a subsequent PR. The common part is expression override map. The two options for sharing code for expression override are:
* BaseTableInfo with the expression override code
* Move expression override to a private helper method.

In the draft PR you had chosen 2nd one. Couple of q's
* Option 1 or Option 2?
* If Option 1, then should we consider adding support to set/unset query options in this PR itself ? 

My concern is that the PR is adding a dummy class and we havent had a chance to discuss the right option to organize the code.